### ISSUE TITLE
Unable to open Select2 combobox on iOS devices when using Voiceover touch screen gestures with Safari

### ISSUE THREAD

**Issue by CraigOcon** [select2/select2#5904]
When using Voiceover touchscreen gestures on iPhone or iPad the Select2 combobox opens and then closes immediately after.

Just in case it was a mistake in my code I have checked with all of the demo examples on the select2.org site and these all exhibit the same problem.

In case you unfamiliar with the Voiceover gestures on a web browser the basic ones are that a single finger flick from left to right on the screen will take you to the next element on the webpage, a single finger flick from right to left will take you to the previous element and a single finger double tap will act as a click on the selected element.

Setting some breakpoints in the js code it appears that the mousedown event is being triggered twice 

this.$selection.on('mousedown', function (evt) {
      // Only respond to left clicks

If I use explore by touch (drag your finger over the screen until you find the element you want to interract with) I can select either the combobox or the readonly textfield of the Select2.  When double tapping on the textfield the mousedown event is only triggered once and the combobox stays open.

Once the combobox is open the rest of it seems to work okay with Voiceove as far as I have tested it, even reading the results from a remote source which doesn't work at the moment when using a screenreader like Jaws or NVDA.

As a quick test for the remote data not reading with a screenreader in the function

  Results.prototype.option = function (data) {

I changed 

  if (data._resultId != null) {
        option.id = data._resultId;
    }

 to

  if (data._resultId != null) {
        option.id = data._resultId;
    }
    else {
        if (data.id != null) {
            data._resultId = 'select2-result-' + data.id;
            option.id = data._resultId;
        }
    }

which then allows the screen readers to read the results as they are moved through in the list (all linked to the 'aria-activedescendant' property)
